stream.go

```diff
@@ -16,19 +18,17 @@ type stream struct {
 
 func (s *stream) Read(b []byte) (n int, err error) {
 	n, err = s.Stream.Read(b)
-	if serr, ok := err.(quic.StreamError); ok && serr.Canceled() {
+	if errors.Is(err, &quic.StreamError{}) {
 		err = mux.ErrReset
 	}
-
 	return n, err
 }
 
 func (s *stream) Write(b []byte) (n int, err error) {
 	n, err = s.Stream.Write(b)
-	if serr, ok := err.(quic.StreamError); ok && serr.Canceled() {
+	if errors.Is(err, &quic.StreamError{}) {
```

Comment:
Eh, actually, a simple `err != nil && errs.Is...` would do it. Allocating on error paths is fine, it would just be nice to avoid the extra one on the "normal" path (where possible).